.github/workflows/toolchain-version-check.yaml

```
@@ -0,0 +1,76 @@
+---
+name: Toolchain Version Check
+"on":
+  schedule:
+    - cron: "0 5 * * 5" # 5am UTC Fridays
+
+jobs:
+  check-toolchain-version:
+    name: Check toolchain version
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4.1.1
+
+      - name: Install Ruby toolchain
+        uses: ruby/setup-ruby@8575951200e472d5f2d95c625da0c7bec8217c42 # v1.161.0
+        with:
+          ruby-version: ".ruby-version"
+          bundler-cache: true
+
+      - name: Read existing rustc version
+        id: rust_version
+        run: |
+          echo "version=$(bundle exec rake toolchain:version)" >> "$GITHUB_OUTPUT"
+
+      - name: Check the latest rustc version
+        id: latest_rust_release
+        uses: octokit/request-action@89697eb6635e52c6e1e5559f15b5c91ba5100cb0 # v2.1.9
+        with:
+          route: GET /repos/{owner}/{repo}/releases/latest
+          owner: rust-lang
+          repo: rust
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract rust version info
+        id: version_info
+        env:
+          CURRENT: ${{ steps.rust_version.outputs.version }}
+          NEXT: ${{ fromJson(steps.latest_rust_release.outputs.data).tag_name }}
+        run: |
+          echo "current=$CURRENT" >> "$GITHUB_OUTPUT"
+          echo "next=$NEXT" >> "$GITHUB_OUTPUT"
+          echo "matches=$(test \"$CURRENT\" = \"$NEXT\" && echo 'true' || echo 'false')" >> "$GITHUB_OUTPUT"
+
+      - name: Update the toolchain version and run sync
+        if: ${{ steps.version_info.outputs.matches != 'true' }}
+        env:
+          NEXT_VERSION: ${{ steps.version_info.outputs.next }}
+        run: |
+          # Matches "RUST_VERSION = 'M.M.P'" and replaces `M.M.P` with `$NEXT_VERSION`
+          sed -iE "s/^\(RUST_VERSION\ =\ '\).*\('\)\$/\1$NEXT_VERSION\2/g" Rakefile
```

This sed invocation looks a lot more understandable to me.

```suggestion
          sed -iE "s/^RUST_VERSION = \"$CURRENT\"$/RUST_VERSION = \"$NEXT\"/g" Rakefile
```

---

For me too :smile: 

I adjusted to single quotes (that we currently have in the Rakefile), and included the other env var changes needed